src/Teams/Services/ManageTeamsMembersService.cs

```diff
@@ -0,0 +1,35 @@
+using Microsoft.EntityFrameworkCore;
+using System.Threading.Tasks;
+using Teams.Data;
+using Teams.Models;
+using Teams.Security;
+
+namespace Teams.Services
+{
+    public class ManageTeamsMembersService : IManageTeamsMembersService
+    {
+        private readonly ICurrentUser _currentUser;
+        private readonly IRepository<Team, int> _teamRepository;
+        private readonly IRepository<TeamMember, int> _memberRepository;
+
+        public ManageTeamsMembersService(IRepository<Team, int> teamRepository, IRepository<TeamMember, int> memberRepository, ICurrentUser currentUser)
+        {
+            _currentUser = currentUser;
+            _teamRepository = teamRepository;
+            _memberRepository = memberRepository;
+        }
+        public async Task<bool> RemoveAsync(int team_id, string member_id)
+        {
+            Team team = _teamRepository.GetByIdAsync(team_id).Result;
+            if (team.TeamOwner == _currentUser.Current.Id() && MemberInTeam(team_id,member_id))
+            {       
+                return await _memberRepository.DeleteAsync(await _memberRepository.GetAll().FirstAsync(t => t.TeamId == team_id && t.MemberId == member_id));
```

Comment:
At least two same calls

Reply:
After you read about *Include* operation you can easily get how to do only two calls, one Get for team and one Delete for member repository